segmentstore/server/src/main/java/io/pravega/segmentstore/server/reading/ContainerReadIndex.java

```diff
@@ -189,8 +190,24 @@ public void triggerFutureReads(Collection<Long> streamSegmentIds) {
                         missingIds.add(segmentId);
                     }
                 }
-            } else {
+                continue;
+            }
+
+            try {
                 index.triggerFutureReads();
+            } catch (ObjectClosedException ex) {
+                // It is possible that between the time we got the pointer to the StreamSegmentReadIndex and when we got
+                // to invoking triggerFutureReads, the StreamSegmentReadIndex has already been closed. If this is the case,
+                // ignore the error.
+                // This is possible in the following scenario: for a Transaction, we have an Append, followed by a Merge;
+                // the Append makes this index eligible for triggering future reads, and the Merge (once committed to Storage)
+                // will close it. If the StorageWriter is sufficiently fast in comparison to the OperationProcessor callbacks
+                // (which could be the case for in-memory unit tests), it may trigger this condition.
+                if (getIndex(segmentId) == null) {
```

Comment:
**Clarification**: if the read index for T1 has been closed, which causes this `ObjectClosedException` to be thrown, wouldn't `getIndex` return null because the index is no longer available? The expected semantics here seems to be that `getIndex` returns the read index for the given segment id even if the read index has been closed.